addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -946,19 +947,35 @@ const Wysiwyg = Widget.extend({
             const proms = [...editableEl.querySelectorAll('.o_modified_image_to_save')].map(async el => {
                 const isBackground = !el.matches('img');
                 el.classList.remove('o_modified_image_to_save');
-                // Modifying an image always creates a copy of the original, even if
-                // it was modified previously, as the other modified image may be used
-                // elsewhere if the snippet was duplicated or was saved as a custom one.
-                const newAttachmentSrc = await this._rpc({
-                    route: `/web_editor/modify_image/${el.dataset.originalId}`,
-                    params: {
-                        res_model: resModel,
-                        res_id: parseInt(resId),
-                        data: (isBackground ? el.dataset.bgSrc : el.getAttribute('src')).split(',')[1],
-                        mimetype: el.dataset.mimetype,
-                        name: (el.dataset.fileName ? el.dataset.fileName : null),
-                    },
-                });
+                const data = (isBackground ? el.dataset.bgSrc : el.getAttribute('src')).split(',')[1];
+                let newAttachmentSrc;
+                if (el.dataset.originalId === 'self') {
```

Comment:
Note that we might have been further:
- Use another new class like `o_b64_image_to_save`
- rename `saveModifiedImages` to something more global (to include both 2 cases)
- make `saveModifiedImages` (which would be renamed) call the current `saveModifiedImages` + a new `saveB64Images`
- The `saveB64Images` method would not need the `isBackground` code as you can't copy or drop a background, right? Plus we would not need the 'self' hack and would leave the current method untouched (which we actually don't share anything)

Don't change it if you think it's not worth / is nitpicking

Reply:
I introduced the `o_b64_image_to_save` class to get rid of the `self` hack, and I renamed `saveModifiedImages` to `savePendingImages`. I did not split in distinct methods though, I just made it handle both kinds of situations.